src/test/java/com/puppycrawl/tools/checkstyle/DetailNodeTreeStringPrinterTest.java

```diff
@@ -100,12 +95,79 @@ public void testCreationOfFakeCommentBlock() throws Exception {
 
     @Test
     public void testNoUnnecessaryTextinJavadocAst() throws Exception {
-        final String actual = DetailNodeTreeStringPrinter.printFileAst(
-                new File(getPath("InputNoUnnecessaryTextInJavadocAst.javadoc")))
-                .replaceAll("\\\\r\\\\n", "\\\\n");
-        final String expected = new String(Files.readAllBytes(Paths.get(
-                getPath("expectedNoUnnecessaryTextInJavadocAst.txt"))), StandardCharsets.UTF_8)
-                .replaceAll("\\\\r\\\\n", "\\\\n");
-        assertEquals(expected, actual);
+        verifyJavadocTree(getPath("expectedNoUnnecessaryTextInJavadocAst.txt"),
+                getPath("InputNoUnnecessaryTextInJavadocAst.javadoc"));
+    }
+
+    @Test
+    public void testUnescapedJavaCodeWithGenericsInJavadoc() throws IOException {
+        try {
+            DetailNodeTreeStringPrinter.printFileAst(new File(getPath(
+                    "InputUnescapedJavaCodeWithGenericsInJavadoc.javadoc")));
+        }
+        catch (IllegalArgumentException ex) {
+            final String expected = getParseErrorMessage(new ParseErrorMessage(
+                    35, MSG_JAVADOC_MISSED_HTML_CLOSE, 7, "parsing"));
+            assertEquals(expected, ex.getMessage());
+        }
+    }
+
+    @Test
+    public void testNoViableAltException() throws IOException {
+        LocalizedMessage.setLocale(Locale.ROOT);
```

Comment:
Done. I had to use key at [javadoc/messages.properties#L3](https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/messages.properties#L3) here to produce error message. This is the same key that is used to produce the original error at [JavadocDetailNodeParser.java#L520](https://github.com/checkstyle/checkstyle/blob/24a748caf712491e04970e1ec992d932c02da830/src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java#L520). This case is similar to the existing case at [AbstractJavadocCheckTest.java#L65](https://github.com/checkstyle/checkstyle/blob/a5abba7c8c7834d88e5c26bc9f54cacfec43e616/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java#L65), which has been modified in this PR. 

Usage of this key results in no translations for phrase `no viable alternative at input` which is arguably a common exception thrown to users. We can look into improvement of this key and other such cases in another issue perhaps, what do you think